library/common-library/templates/_secret.yaml.tpl

```diff
@@ -0,0 +1,87 @@
+{{/*
+Return local licenseKey if set, global otherwise.
+This helper is for internal use.
+*/}}
+{{- define "common.licenseKey._licenseKey" -}}
+{{- if .Values.licenseKey -}}
+  {{- .Values.licenseKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.licenseKey -}}
+    {{- .Values.global.licenseKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name of the secret holding the License Key.
+This helper is for internal use.
+*/}}
+{{- define "common.licenseKey._customSecretName" -}}
+{{- if .Values.customSecretName -}}
+  {{- .Values.customSecretName -}}
+{{- else if and .Values.global -}}
+  {{- if .Values.global.customSecretName -}}
+    {{- .Values.global.customSecretName -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name of the secret holding the License Key.
+*/}}
+{{- define "common.licenseKey.secretName" -}}
+{{ include "common.licenseKey._customSecretName" . | default (printf "%s-license" (include "common.naming.fullname" . )) }}
+{{- end -}}
+
+{{/*
+Return the name key for the License Key inside the secret.
+This helper is for internal use.
+*/}}
+{{- define "common.licenseKey._customSecretKey" -}}
+{{- if .Values.customSecretLicenseKey -}}
+  {{- .Values.customSecretLicenseKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.customSecretLicenseKey }}
+    {{- .Values.global.customSecretLicenseKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name key for the License Key inside the secret.
+*/}}
+{{- define "common.licenseKey.secretKeyName" -}}
+{{ include "common.licenseKey._customSecretKey" . | default "licenseKey" }}
+{{- end -}}
+
+{{/*
+Charts using this library can implement this function. If it returns a truty value, common.licenseKey.secret will not
```

Comment:
I was thinking about https://github.com/newrelic/helm-charts/pull/543, where the licenseKey is optional :)

Reply:
Therefore I always sponsor to add feature whenever we actually need them, otherwise there is the risk of having this there for ever and never actually using it

Reply:
Fair point, I've removed the feature for now :)